runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -54,7 +55,13 @@ public void incrementLapCount(Client client)
 
 		int currentExp = client.getSkillExperience(Skill.AGILITY);
 		int nextLevel = client.getRealSkillLevel(Skill.AGILITY) + 1;
-		int remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
+
+		int remainingXp;
+		do
+		{
+			remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
+			nextLevel++;
+		} while (remainingXp < 0);
```

Comment:
What is this for? The xp is being added before the level is increased?

Reply:
I'm not sure I understand the question, but it happens that getSkillExperience returns more exp than what is the exp of getRealSkillLevel + 1 this happens if the player levels up when completing the lap.
Next level starts at getRealSkillLevel + 1 before the loop and is increased until either the level is more than currentExp that is gotten from getSkillExperience or the level is greater or equal than MAX_VIRT_LEVEL.